### DISCUSSION TITLE
강동훈 스토리

### DISCUSSION THREAD

**Discussion by ypd01018** [ondal-house/zu-story#17] [스토리 작성 안건]
UI/UX의 경우 당근마켓을 참조

## 프로젝트 제목
- 주담
  - 주택 담보대출 아님. 주류 담화..ㅎㅎ
- 블랙아웃
  - 후기 글을 쓴 시점은 필름 나갔을 것 같아서
- 드렁큰 라이트
  -  글 쓰는 시점에 취해있을 것 같음
- 드렁큰 OO
  - OO는 아무거나..ㅎㅎ 드렁큰 맘에 뭔가 듬
- 주스토리
  - 술 이야기..ㅎ
  
## 로그인 페이지
- 소셜 로그인/회원가입

## 가입 플로우
- 소셜 인증 및 정보제공 동의
- 닉네임 설정

## Nav Bar(웹) / Tab Bar(앱)
- 홈
- 제품 공간
- 주종 공간
- 채팅 공간
- 내 페이지

## 홈
- Hot한 제품
    - 특정 기간동안(어제/지난 주 등) 방문자가 많은 아이템
    - 상위 10개정도 노출?
- 연령 별 선호 제품
- 인기 글
    - 주종 별 커뮤니티에서 작성된 글 중 따봉/조회수가 많은 N개 글
- 신규 글
    - 주종 별 커뮤니티에 작성된 최신 N개 글 조회
- 여기는 제일 나중에 기능 다 넣고 생각해도 될 듯 ㅋㅋ

## 제품 공간
- 당근마켓 동네 정보처럼 상단에 필터기능 부여
- 필터에서 주종 선택 가능
- 연령 선호도 / 평균 별점 / 조회 수 / 리뷰 수 sort 가능
- 제품 Read는 당근마켓의 중고거래(홈)처럼 UI 만들면 좋을 듯(제목 / 내용)
- 제품 추가 요청 버튼

### 제품 후기
- 피드 형식으로 제공(당마 동네 생활)
- 제목x / 별점/내용/이미지/ 그 외 상세 후기 o
- 제품 후기에도 댓글이 있으면 좋지 않을까? 무신사처럼

## 주종 공간
- 당마 동네생활 UI/UX가져오자
- 상단 주종 필터
- Q&A / 잡담 등 필터
- 제목은 빼는게 꽤 멋있는 것 같다
- 분류/내용/이미지/댓글

## 채팅
- 특정 사용자와 채팅 기능 제공
- 솔직히 그렇게 메인 기능은 아무리 봐도 아닌 것 같음
- 기술적 니즈 / 넣을 게 없어서 넣어봄

## 마이 페이지
- 내 정보 표시
- 정보 수정
- 북마크?
- 키워드 모아보기?
- 제품 추가 요청
- 넣을게 딱히는 없는 듯. 나중에 붙이고 싶은거 있으면 좋을 듯

# 스토리
- 스토리 짜면서 든 생각 -> 위 정도만 정해놓고 개발 기술 스택 선정이 먼저 아닌가?
- 전체적인 그림 그리려니 머리 안굴러감. 원래 당장 할 수 있다고 느껴지는 정도만 하는게 무력감을 느끼지 않는 방법이라 들었음 ㅋㅋ 그러니 당장 필요한것만 적을랭
- 참여 인원 분들의 흥미 분야가 궁금한 (FE / BE / 데봅,인프라/ DE)
## 인프라 / 그 외
- CI / CD 간단하게만 구축
- 도커 설정

## FE
- FE 프레임워크 세팅
- 글로벌 레이아웃 세팅

- 회원가입 / 로그인

- 제품 Read
- 제품 Filter
- 제품 후기 CRUD

- 위 스토리에 필요한 퍼블리싱 (이게 제일 힘들어보임)
## BE
- BE 프레임워크 세팅
- 테이블 설계
- 로그인 / 회원 가입
- 제품 Read
- 제품 후기 CRUD

# 개발 스택
## FE
- 앱/웹 선택이 가장 먼저로 보임
- 웹 -> React
- 앱 -> Android/IOS 네이티브 OR RN OR 플러터
- 이걸 먼저 정해야 이야기가 되겟네요 ㅋㅋㅋ

## BE
- Spring Boot
    - 사실 BE에서 많은 프레임워크/ 언어가 존재하고 어떤 것으로 만들든 목표하는 개발 결과물에 문제가 없다고 다들 알 것임
    - 익스프레스 / 쟝고 등 많음
    - 많은 기업에서 자바/스프링 부트를 사용하는 이유
        - 속도 및 정적언어/OOP의 안정성
        - 가장 자료가 많음
        - 가장 큰 이유 -> 인력 시장에서 스프링 부트가 가장 많음

    - 사실 결과적으로 잘 만들고 싶다와 이걸 나중에 스펙으로 써먹고 싶다면 스프링부트만한게 없음....

- Swagger VS Rest Dosc 
    - ~이건 나도 잘 안알아봐서 봐야함~
    - 살펴본 결과 테스트와 문서화를 지향하는 rest docs vs 실행가능한 swagger 선택지임
    - 뭔가 난 둘다 하고싶음
    - (https://luvstudy.tistory.com/186)(https://jwkim96.tistory.com/274)사례가 있네? 둘다 적용하면 좋지 않을까?
  
- MySQL
- Spring Security?
    - 인증 관련해서 담당할 수 있는 것
    - 고민을 해봐야함
    - 러닝 커브가 꽤 높음
        - 인터셉터에서 해결함으로 간단히 할 수 있는 것을 많은 시간이 들 수 있음
        - 인터셉터에 많은 것을 넣지 않는다면 과한 공수..?투입이 아닐까?
        - 그러나 사용해봄으로서 학습 경험을 만들 수 있다는 것은 좋게 보임

- JUnit5
    - 그리고 Jupiter

- JPA
    - OOP를 위해서라면 ORM인 JPA를 쓰는게 좋을것같아요.

- SonarQube
    - BE 코드의 퀄리티를 높입시당.

- Flyway
    - 이거만한게 없지

- ElasticSearch?
    - 검색엔진으로 사용 가능
    - 시간이 많이 남는다면 나중에 써보면 좋을 듯 ㅋㅋ

- Redis
    - 세션 관리용
    - 추후 채팅 기능 넣는다면 채팅 뿌려주기 Pub/Sub을 위해 필요

- Java?Kotlin?
### 자바
- 장점
    - 익숙함
    - 추가적인 언어의 러닝커브가 없으므로 조금 더 비즈니스 로직 / OOP / 스프링 부트 정석 활용에 집중 할 수 있음
- 단점
    - 너무 강압적임
    - 주저리주저리 항상 메서드 및 코드가 너무 김
    - 그놈의 NPE

### 코틀린
- 장점
    - 자바 학습 이력이 있으면 쉽게 학습 가능 및 JVM 호환으로 스프링 부트 그대로 가능
    - 인프런 들어봤는데 확실히 코드 간결함
    - 카카오에서 발표 결과 자바 코드에 비해 코드 양이 66% 줄었다고 함
    - 모던한 언어임(마치 JS -> TS 흐름과 유사함)
    - 최근 많은 기업에서 자바 -> 코틀린으로 넘어가고 있음
    - 널 세이프티

- 단점
    - 배워야함
    - 쉽다고는 하긴 하는데 그럼에도 만약 이것의 사용을 어려워하는 개발자가 있을 경우, 목표한 BE를 한번 잘 써보고 싶다까지 못가고 언어 사용에서 허우적거리다가 좋은 코드를 못 만들수 있음

## Cloud / Infra
- 세훈이가 EC2 쓰는 것 보다 ECS(맞나? 어쨋든 컨테이너) 쓰는게 더 싸대요.
- Jenkins든 GitHub Action든 Travis CI든
- docker
- 세훈 기용이가 잘 해주겠지...잘 알려주겠지...

**Comment by haneeeul** [comment]
#### 제목 x ?!
-> 제목이 없는 경우는 해당 게시글에 대한 상세 페이지 or 제목을 기준으로 하는 아코디언 UI 없이 간다는 것인지 정할 필요가 있어 보입니다.

#### 스토리에 필요한 퍼블리싱
-> 사실상 FE 최대 난관
-> 자세한 건 회의에서 이야기 하겠지만 일단 UI 정의를 위한 피그마/제플린(이건 우리가 선택하기 참 힘듦. 우리 중에 포토샵/일러스트 능력자 있나요? ㅠㅡㅠ) 등 하다못해 ppt 라도 무조건 하나의 디자인 파일은 필요하다고 생각합니다. 일단 제 추천은 피그마인데,  
1. 온라인 지원
2. 플러그인이 상당히 다양
  i. 유료 플러그인 사용 시 생각보다 쉽고 빠르게 디자인 파일 + 퍼블리싱 결과물(ex. Anima) 뽑아낼 수 있음.

**Reply by ypd01018** [reply]
### 제목 x
트위터/당마처럼 생각했어요 ㅋㅋㅋ
<img src="https://user-images.githubusercontent.com/37091190/193575479-8938adaf-40d5-48c9-bbff-cf1f0b5a3686.jpeg" width="50%" height="50%"/>

이런거!

### 퍼블리싱
- 이거 진짜 이슈임
- 공수 진짜 너무 많이 듬
- 저도 피그마밖에 몰라서 찬성이긴해요...다른 걸 선택할 역량 지식이 없음..................
